core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -611,6 +611,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
         AppInfoParser.unregisterAppInfo(jmxPrefix, config.brokerId.toString)
         shutdownLatch.countDown()
         info("shut down completed")
+      } else {
+        isShuttingDown.set(false)
```

Comment:
@ijuma See my adjustment. I think just flipping the conditional like you did suffices. No need for synchronization due to the ordering of latch and atomicBoolean operations imo.